Documentation/security/network/encryption-ipsec.rst

```diff
@@ -21,6 +21,23 @@ Packets are not encrypted when they are destined to the same node from which
 they were sent. This behavior is intended. Encryption would provide no benefits
 in that case, given that the raw traffic can be observed on the node anyway.
 
+v1.18 Encrypted Overlay
+=========================
+Prior to v1.18, IPsec encryption was performed before tunnel encapsulation.
+From Cilium v1.18 and forward, Cilium's IPsec encryption datapath will send
+traffic for overlay encapsulation prior to IPsec encryption when tunnel mode is
+enabled.
+
+With this change, the security identities used for policy enforcement are
+encrypted on the wire. This is a security benefit.
+
+A disruption-less upgrade from v1.17 to v1.18 can only be achieved by fully
+patching v1.17 to its latest version. Migration specific code was added to
+newer v1.17 releases to support a disruption-less upgrade to v1.18.
+
+Once patched to the newest v1.17 stable release, a normal upgrade to v1.18 can
+be performed.
```

Comment:
It feels like it should be also noted in the upgrade notes (`Documentation/operations/upgrade.rst`), wdyt?

Reply:
Yup, I looked over this document just now. 

Its pretty busy already but I will add a call out or note saying "If upgrading to v1.17 -> v1.18 and using IPsec please read the caveats in {link to documentation section}"

Reply:
@lambdanis I added a note in the upgrade documentation.